File: frontend/src/types/task.ts
export interface HumanTask {
  id: string;
  workflowExecutionId: string;
  workflowName: string;
  nodeId: string;
  title: string;
  description: string;
  priority: 'low' | 'medium' | 'high' | 'critical';
  status: 'pending' | 'claimed' | 'completed' | 'escalated';
  assignedTo?: string;
  claimedBy?: string;
  context: TaskContext;
  options: TaskOption[];
  deadline?: string;
  createdAt: string;
  updatedAt: string;
}

export interface TaskContext {
  originalDocument?: string;
  processState: Record<string, any>;
  botActions: BotAction[];
  screenshots: string[];
  relevantData: Record<string, any>;
}

export interface BotAction {
  timestamp: string;
  action: string;
  target: string;
  result: string;
  confidence: number;
}

export interface TaskOption {
  id: string;
  label: string;
  description: string;
  impact: 'low' | 'medium' | 'high';
  confidence: number;
  reasoning: string;
}

File: backend/app/schemas/task.py
from pydantic import BaseModel, Field
from typing import Optional, Dict, Any, List
from datetime import datetime
from uuid import UUID

class TaskCreate(BaseModel):
    title: str = Field(min_length=1, max_length=500)
    description: Optional[str] = None
    priority: str = "medium"
    assigned_to: Optional[UUID] = None
    context: Optional[Dict[str, Any]] = Field(default_factory=dict)
    options: Optional[List[Dict[str, Any]]] = Field(default_factory=list)

class TaskUpdate(BaseModel):
    status: Optional[str] = None
    assigned_to: Optional[UUID] = None

class TaskResponse(BaseModel):
    id: UUID
    tenant_id: UUID
    workflow_execution_id: Optional[UUID]
    title: str
    description: Optional[str]
    priority: str
    status: str
    assigned_to: Optional[UUID]
    context: Dict[str, Any]
    options: List[Dict[str, Any]]
    created_at: datetime
    updated_at: datetime
    
    class Config:
        from_attributes = True